### PR TITLE
Added test cases for the DynamoDb connection and one Categories filter

### PR DESCRIPTION
**Pull request by shub6691**
These test cases are for the testing purpose.  More test cases need to be added and code refactoring is required.

### CODE REVIEW

File: adapters/dynamoDB_adapters/pom.xml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        <version>1.2.35-SNAPSHOT</version>
+        <relativePath>../../pom.xml</relativePath>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+
+    <groupId>org.atomhopper.adapter</groupId>
+    <artifactId>dynamoDB_adapters</artifactId>
+    <packaging>jar</packaging>
+
```
Comment (AJStieren) [inline adapters/dynamoDB_adapters/pom.xml:17]: Add a `name` attribute matching the format of the other adapters. Example [link](https://github.com/rackerlabs/atom-hopper/blob/37f01063cec90c54b260e7caefa238f7a62d3a71/adapters/jdbc/pom.xml#L16).

File: adapters/dynamoDB_adapters/src/main/java/org/atomhopper/dynamodb/adapter/DynamoFeedPublisher.java [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+import org.atomhopper.util.uri.template.URITemplate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.yammer.metrics.core.Counter;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.text.Format;
+import java.text.SimpleDateFormat;
+import java.util.*;
```
Comment (AJStieren) [inline adapters/dynamoDB_adapters/src/main/java/org/atomhopper/dynamodb/adapter/DynamoFeedPublisher.java:31]: Make imports explicit

File: adapters/dynamoDB_adapters/src/main/java/org/atomhopper/dynamodb/adapter/DynamoFeedPublisher.java [hunk 3]
```diff
@@ -0,0 +1,245 @@
+package org.atomhopper.dynamodb.adapter;
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+import com.amazonaws.services.dynamodbv2.document.*;
```
Comment (AJStieren) [inline adapters/dynamoDB_adapters/src/main/java/org/atomhopper/dynamodb/adapter/DynamoFeedPublisher.java:5]: Make imports explicit

File: adapters/dynamoDB_adapters/src/main/java/org/atomhopper/dynamodb/model/PersistedEntry.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package org.atomhopper.dynamodb.model;
+import com.amazonaws.services.dynamodbv2.datamodeling.*;
+import lombok.Data;
+import java.util.*;
```
Comment (AJStieren) [inline adapters/dynamoDB_adapters/src/main/java/org/atomhopper/dynamodb/model/PersistedEntry.java:4]: Make imports explicit

File: adapters/dynamoDB_adapters/src/main/java/org/atomhopper/dynamodb/query/SearchToSqlConverter.java [hunk 5]
```diff
@@ -0,0 +1,311 @@
+package org.atomhopper.dynamodb.query;
+
+import com.unboundid.ldap.sdk.Filter;
+import com.unboundid.ldap.sdk.LDAPException;
+import org.apache.commons.lang.StringUtils;
+
+import java.util.*;
```
Comment (AJStieren) [inline adapters/dynamoDB_adapters/src/main/java/org/atomhopper/dynamodb/query/SearchToSqlConverter.java:7]: Make imports explicit

File: adapters/dynamoDB_adapters/src/main/java/org/atomhopper/dynamodb/query/SqlBuilder.java [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+import org.apache.commons.lang.StringUtils;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+
+import java.util.List;
+import java.util.Map;
+
+public class SqlBuilder {
```
Comment (AJStieren) [inline adapters/dynamoDB_adapters/src/main/java/org/atomhopper/dynamodb/query/SqlBuilder.java:13]: It is _so_ weird that you can use SQL based queries for DynamoDB as a primary strategy. This isn't a request to chagne it, just a comment on... it's weird.

File: adapters/dynamoDB_adapters/src/test/java/DynamoDBFeedPublisherTest.java [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.springframework.http.HttpStatus;
+
+import java.text.Format;
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+import static junit.framework.Assert.*;
+import static org.mockito.Mockito.*;
```
Comment (AJStieren) [inline adapters/dynamoDB_adapters/src/test/java/DynamoDBFeedPublisherTest.java:28]: Make imports explicit

File: adapters/dynamoDB_adapters/src/test/java/DynamoDBFeedSourceTest.java [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+import org.atomhopper.dynamodb.adapter.DynamoFeedPublisher;
+import org.atomhopper.dynamodb.model.PersistedEntry;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.runners.Enclosed;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import java.util.*;
```
Comment (AJStieren) [inline adapters/dynamoDB_adapters/src/test/java/DynamoDBFeedSourceTest.java:18]: Make imports explicit

File: adapters/dynamoDB_adapters/src/test/java/DynamoDBFeedSourceTest.java [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.*;
```
Comment (AJStieren) [inline adapters/dynamoDB_adapters/src/test/java/DynamoDBFeedSourceTest.java:22]: Make imports explicit

**Review by AJStieren — CHANGES_REQUESTED** [review]
There are 2 big ticket items off the top:

* The tests do not pass
* There is an excessive amount of commented/non-functional code

Also the keys shouldn't be populated here.

**Comment by AJStieren** [comment]
This should not be a PR to Master until the adapter feature has been completed

**Comment by shub6691** [comment]
Closing the PR and raising the fresh PR with the removal of commented codes

**Pull request closed without merging**